Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -824,20 +862,20 @@ namespace Terrain
 
             const auto jobLambda = [this, sector, gridMeters]() -> void
             {
-                AZStd::vector<HeightDataType> meshHeights;
-                AZStd::vector<NormalDataType> meshNormals;
+                AZStd::vector<HeightNormalVertex> meshHeightsNormals;
 
                 {
                     SectorDataRequest request;
                     request.m_samplesX = GridVerts1D;
                     request.m_samplesY = GridVerts1D;
                     request.m_worldStartPosition = AZ::Vector2(sector->m_worldX * gridMeters, sector->m_worldY * gridMeters);
                     request.m_vertexSpacing = gridMeters / GridSize;
+                    request.m_useVertexOrderRemap = true;
```

Comment:
Is this just for testing, or are there cases where the vertex order *isn't* remapped?

Reply:
Actually more cases _aren't_ remapped than are, which is why it defaults to false. It's only remapped for the primary heights/ normals. The lod heights / normals need some extra processing so they're gotten in linear order first, processed, then saved in remapped order. The ray tracing stuff also doesn't use the remapped vertices (although I consider all of that code to be temporary).

In the future we may be able to just use remapped coords all the time - I don't like having that `if` check in the for loop. but I'll keep the option for now.